### ISSUE TITLE
Keywords failing to function correctly within filters

### ISSUE THREAD

**Issue by DrCode32** [mastodon/mastodon#31352]
### Steps to reproduce the problem

1. Go to the filter settings and create a new filter with the keyword "Test".
2. Go home and you will see that the post with "Test0" is not filtered.

### Expected behaviour

Words that contain filtered keywords or specific characters should not be filtered.

### Actual behaviour

Words that contain a part matching the filtered keyword or have a specific character in them are not filtered.

### Detailed description

Some people intentionally omit spaces between words to circumvent filters. This causes filters to malfunction. For instance, if your filter contains the word "Test", a post with "Test0" will not be filtered. Additionally, some individuals insert special characters between filtered words to bypass filters. For example, the character: "‌"

![Screenshot_20240809_114117_Chrome](https://github.com/user-attachments/assets/e6984669-d260-4d1d-84e6-588f4d642e3a)

### Mastodon instance

any instance - khiar.net

### Mastodon version

v4 and up - v4.2.10

### Browser name and version

any browser - Google Chrome 127.0.6533.84

### Operating system

any OS - Android 14

**Comment by ClearlyClaire** [comment]
Please click “Edit filter” so we can see whether “Whole word” is checked. The matching you suggest is already supposed to work, but it is optional and has to be enabled per-filter.

**Comment by Gargron** [comment]
Also I'm not sure if filters are supposed to work on your own posts.

**Comment by ClearlyClaire** [comment]
Ah, yes, filters are not supposed to work on your own posts.

**Comment by DrCode32** [comment]
> Please click “Edit filter” so we can see whether “Whole word” is checked. The matching you suggest is already supposed to work, but it is optional and has to be enabled per-filter.

‌Don't work.

**Comment by ThisIsMissEm** [comment]
Could this be another of the ß not matching cases? I'm just looking at the examples and both have RTL text (arabic?) in them, perhaps that's causing an issue like ß next to hashtags did?

**Comment by DrCode32** [comment]
> Could this be another of the ß not matching cases? I'm just looking at the examples and both have RTL text (arabic?) in them, perhaps that's causing an issue like ß next to hashtags did?

I don't think it has anything to do with it. There is also a problem with English posts.

**Comment by ThisIsMissEm** [comment]
In which case I'm guessing you're using "Whole Word" as a checkbox when you edit that filter (perhaps the name of that isn't clear enough)

When "Whole Word" is checked, the filter of "test" will not match "test1", but should match "test, foo"

When "Whole Word" isn't checked, it'll match "test1", "abctest", "testabc" and "abctestabc", as examples.
